crates/bevy_ecs/src/bundle.rs

```diff
@@ -161,8 +161,12 @@ pub unsafe trait Bundle: Send + Sync + 'static {
 
     /// Calls `func` on each value, in the order of this bundle's [`Component`]s. This passes
     /// ownership of the component values to `func`.
+    ///
+    /// # Safety
+    ///
+    /// The passed in `StorageType` needs to be correct for the provided component
```

Comment:
This safety comment isn't clear enough: different components within the same bundle may have different storage types.

Reply:
Actually unsure how to handle this. The safety constraint is on the implementor, not the caller. Might move this invariant onto the unsafe trait impl instead of the function.

Reply:
Yeah, is a detail of the trait. IMO we can make this method safe.

Reply:
The type level docs basically say "don't". I'll leave it as internal safety comments on `get_components` then.